src/Symfony/Component/HttpKernel/ControllerMetadata/Argument/ArgumentMetadata.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\ControllerMetadata\Argument;
+
+/**
+ * Responsible for storing metadata of an argument.
+ *
+ * @author Iltar van der Berg <kjarli@gmail.com>
+ */
+final class ArgumentMetadata implements ArgumentMetadataInterface
```

Comment:
In any case, you don't need an interface for that. Instead of storing a class, why not just store the serialized object?